core/internal/runfiles/uploader.go

```diff
@@ -121,8 +122,16 @@ func (u *uploader) Process(record *spb.FilesRecord) {
 		}
 		runPath := *maybeRunPath
 
-		u.knownFile(runPath).
-			SetCategory(filetransfer.RunFileKindFromProto(file.GetType()))
+		category := filetransfer.RunFileKindFromProto(file.GetType())
+		// Special case handling for media files, which are currenlty all
+		// stored under the "media" directory. If the file is already assign the
+		// non-default category (that is not `Other`), we don't override it.
+		// TODO: revisit this once we rewrite the media code on the client side.
```

Comment:
* Please double check the spelling and grammar
* The TODO isn't actionable, it will rot. Better to remove it.
* Consider rewriting for conciseness

I'd reduce the comment to just:

> Files under media/ always count as media files, but the client does not always mark their type.